UI-Side Compositing : Text on a Button in PDF and inputted text appear upside down

rr-codes opened the pull request:
#### 9358d2aa116959580bc484002179a2cc3e133698
<pre>
UI-Side Compositing : Text on a Button in PDF and inputted text appear upside down
<a href="https://bugs.webkit.org/show_bug.cgi?id=255797">https://bugs.webkit.org/show_bug.cgi?id=255797</a>
rdar://107127603

Reviewed by Megan Gardner.

With UI-side compositing, PDF annotations were rendered upside-down due to `isGeometryFlipped`
being `true` for their layers.

This PR fixes this by adopting a PDFKit delegate method on the PDF layer controller&apos;s delegate
to access each annotation&apos;s layers, and then flips them vertically with a transform to get them
to be the correct orientation.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(-[WKPDFLayerControllerDelegate initWithPDFPlugin:]):
(-[WKPDFLayerControllerDelegate pdfLayerController:didUpdateLayer:forAnnotation:]):
(WebKit::PDFPlugin::setView):

Canonical link: <a href="https://commits.webkit.org/263343@main">https://commits.webkit.org/263343@main</a>
</pre>

megangardner reviewed (approved):
Any way to write a test for this?

rr-codes committed:
UI-Side Compositing : Text on a Button in PDF and inputted text appear upside down
https://bugs.webkit.org/show_bug.cgi?id=255797
rdar://107127603

Reviewed by Megan Gardner.

With UI-side compositing, PDF annotations were rendered upside-down due to `isGeometryFlipped`
being `true` for their layers.

This PR fixes this by adopting a PDFKit delegate method on the PDF layer controller's delegate
to access each annotation's layers, and then flips them vertically with a transform to get them
to be the correct orientation.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(-[WKPDFLayerControllerDelegate initWithPDFPlugin:]):
(-[WKPDFLayerControllerDelegate pdfLayerController:didUpdateLayer:forAnnotation:]):
(WebKit::PDFPlugin::setView):

Canonical link: https://commits.webkit.org/263343@main
